.github/workflows/trivy.yml

```diff
@@ -0,0 +1,87 @@
+---
+#
+# This workflow runs Trivy on a Docker image
+# It can pull the image from a container registry
+# or download a tar file.  The latter is used
+# to check a container image prior to publishing
+# to the registry.
+
+name: Run Trivy on a Docker image and push results to GitHub
+
+on:
+  workflow_call:
+    inputs:
+      NOTEBOOK_TYPE:
+        required: true
+        type: string
+      SOURCE_TYPE: # 'tar' or 'image'
+        required: true
+        type: string
+      TARFILE_NAME: # only used if SOURCE_TYPE=='tar'
+        required: false
+        type: string
+      IMAGE_NAME:
+        required: true
+        type: string
+      EXIT_CODE:
+        required: false
```

Comment:
doesn't this need a default value?

Reply:
No: With no 'default value' the default value is `null` which, for the Trivy, `exit-code` is zero (its default).

Reply:
it's not clear that this would be `null` since this value's type is set to `number`.  I would consider setting a default value of `0` for this.

Reply:
Yes, I agree it's not obvious that null/0 is the default.  I added 0 as the default and added a comment to explain what 0 means.